crates/taplo-cli/src/commands/queries.rs

```diff
@@ -14,6 +17,11 @@ impl<E: Environment> Taplo<E> {
     pub async fn execute_get(&self, cmd: GetCommand) -> Result<(), anyhow::Error> {
         let mut stdout = self.env.stdout();
 
+        // `--separator` should only be handled for a text output for mat
+        if cmd.separator.is_some() && !matches!(cmd.output_format, OutputFormat::Value) {
```

Comment:
I figured making `cmd.separator` an `Option<String>` would catch instances where the end user provided a separator for non text formats.

I wasn't able to find a "requires other arg to be this value" in clap to guard against this:
https://docs.rs/clap/latest/clap/struct.Arg.html